projectional/src/main/java/jetbrains/jetpad/projectional/selection/Selection.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2012-2016 JetBrains s.r.o
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.jetpad.projectional.selection;
+
+import jetbrains.jetpad.cell.Cell;
+
+public class Selection {
+  public final Cell start;
+  public final Object startPosition;
+  public final Cell end;
+  public final Object endPosition;
+
+  public Selection(Cell start, Object startPosition, Cell end, Object endPosition) {
```

Comment:
What will we have in startPosition and endPosition? Objects of what class? Why is it Object?